src/terms/local.jl

```diff
@@ -69,23 +69,29 @@ Atomic local potential defined by `model.atoms`.
 """
 struct AtomicLocal end
 function (::AtomicLocal)(basis::PlaneWaveBasis{T}) where {T}
-    model = basis.model
-
     # pot_fourier is <e_G|V|e_G'> expanded in a basis of e_{G-G'}
     # Since V is a sum of radial functions located at atomic
     # positions, this involves a form factor (`local_potential_fourier`)
     # and a structure factor e^{-i G·r}
-
-    pot_fourier = map(G_vectors(basis)) do G
-        pot = sum(model.atom_groups) do group
-            element = model.atoms[first(group)]
-            form_factor::T = local_potential_fourier(element, norm(model.recip_lattice * G))
-            form_factor * sum(cis2pi(-dot(G, r)) for r in @view model.positions[group])
+    model = basis.model
+    G_frac = G_vectors(basis)
+    G_cart = G_vectors_cart(basis)
+
+    form_factors = Dict{Tuple{Int,T},T}()
+    pot_fourier = Array{Complex{T}, 3}(undef, size(G_frac))
```

Comment:
Please keep the `map` construct as this was introduced to help with AD and GPUs.